src/platform/nrfconnect/wifi/ConnectivityManagerImplWiFi.cpp

```diff
@@ -39,124 +39,200 @@ using namespace ::chip::TLV;
 namespace chip {
 namespace DeviceLayer {
 
-ConnectivityManager::WiFiStationMode ConnectivityManagerImplWiFi::_GetWiFiStationMode(void)
+CHIP_ERROR ConnectivityManagerImplWiFi::InitWiFi()
 {
-    return ConnectivityManager::WiFiStationMode::kWiFiStationMode_Disabled;
+    return WiFiManager::Instance().Init();
 }
 
-CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode val)
+ConnectivityManager::WiFiStationMode ConnectivityManagerImplWiFi::_GetWiFiStationMode(void)
 {
+    if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+    {
+        mStationMode = (WiFiManager::StationStatus::DISABLED == WiFiManager().Instance().GetStationStatus())
+            ? ConnectivityManager::WiFiStationMode::kWiFiStationMode_Disabled
+            : ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled;
+    }
+    return mStationMode;
+}
+
+CHIP_ERROR ConnectivityManagerImplWiFi::_SetWiFiStationMode(ConnectivityManager::WiFiStationMode aMode)
+{
+    VerifyOrReturnError(ConnectivityManager::WiFiStationMode::kWiFiStationMode_NotSupported != aMode,
+                        CHIP_ERROR_UNSUPPORTED_CHIP_FEATURE);
+
+    if (aMode != mStationMode)
+    {
+        mStationMode = aMode;
+        if (mStationMode != ConnectivityManager::WiFiStationMode::kWiFiStationMode_ApplicationControlled)
+        {
+            bool doEnable{ ConnectivityManager::WiFiStationMode::kWiFiStationMode_Enabled == mStationMode };
+            if (doEnable)
+            {
+                OnStationConnected();
+            }
+            else
+            {
+                WiFiManager().Instance().DisconnectStation();
+                OnStationDisconnected();
```

Comment:
I think eventually we should call it from some WPA supplicant callback. It's worth adding a TODO for that.

Reply:
Yes, I guess there will be some refactoring needed when the callback API is delivered.